wordpress-mobile/WordPress-Android
# gradle: the detekt configuration is now applied by the plugin a8c-app…

The detekt configuration is now applied by the plugin a8c-apply-style-gradle-plugin

This is a proof of concept that shows how we can apply a plugin with custom
common defaults and configuration files into a project.

At first we're applying the detekt plugin and the configuration is taken
care by new plugin a8c-apply-style-gradle-plugin [1]

At first you should clone the plugin's code as the plugin is not published
anywhere:

  git clone git@github.com:potyl/apply-style-gradle-plugin.git

It is recommended to clone the repo as a sibling of this project.

You can then apply the plugin with:

  ./gradlew --include-build ../apply-style-gradle-plugin WordPress:detekt

[1] https://github.com/potyl/apply-style-gradle-plugin

To test:

Run the `detekt` task:

```
./gradlew --include-build ../apply-style-gradle-plugin WordPress:detekt
```
It should run successfully.

**NOTE** this is a draft as the plugin is not published and for now it has to be applied manually to the project.

## Regression Notes
1. Potential unintended areas of impact
The configuration file is completely simplified (copied from woocommerce project) and is now:

```
build:
  maxIssues: 0
```
So far the behavior of detekt is the same so it seems to work well.

Because the plugin is not published I can't test the whole CI.

I also noticed that If I update the project to detekt 1.17.1 things break.
Although this is most likely an issue with the project as the same failure exists in the `develop` branch.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
